src/redux/data.ts

```diff
@@ -684,15 +719,28 @@ export const transactionsReceived = (
 
   const isCurrentAccountAddress =
     accountAddress === getState().settings.accountAddress;
-  if (!isCurrentAccountAddress) return;
+  if (!isCurrentAccountAddress) {
+    loggr.debug(
+      'transactionsReceived: transaction accountAddress does not match current accountAddress',
+      {
+        transactionAccountAddress: accountAddress,
+        currentAccountAddress: getState().settings.accountAddress,
+      }
+    );
+    return;
+  }
 
   if (appended && potentialNftTransaction) {
     setTimeout(() => {
       dispatch(uniqueTokensRefreshState());
     }, 60000);
   }
+
+  const maybeUpdatedPendingTransactions = await maybeFetchF2CHashForPendingTransactions(
+    cloneDeep(pendingTransactions)
+  );
```

Comment:
This checks if any of the pending transactions might be from Ratio. If they are, it fetches the transaction has and mutates the tx itself. That way, the `maybeUpdatedPendingTransactions` appear as if we already had the hash and get filtered out using the functionality that was already in place.

The clone was necessary because values returned from AsynStorage are frozen and can't be mutated.